src/components/pill-selector-dropdown/PillSelectorDropdownField.js

```diff
@@ -85,6 +88,16 @@ class PillSelectorDropdownField extends React.PureComponent<Props, State> {
         onChange(this.createFakeEventTarget(name, options));
     };
 
+    handleParseItems = (inputValue: string): ?Array<Option> => {
+        const { field, inputParser, options } = this.props;
+        const { value: selectedOptions } = field;
+
+        if (inputParser) {
```

Comment:
I would add inputParser to defaultProps, defaulting to `mapInputValuesToOptions` so that here you dont need a if / else block. And probably rename `mapInputValuesToOptions` to something like `defaultInputParser`.

Reply:
Good idea, will change this

Reply:
@priyajeet sorry, I just remembered why I did it like this. `inputParser` and `parseItems` have different signatures. I added `options` and `selectedOptions` to `inputParser` in order to allow greater extensibility. We need this extra function in order to pass the extra parameters to `inputParser`

Reply:
Had misunderstood your point. This has been updated now